### PR TITLE
stylo: Propagate quirks mode information from Gecko to Servo

### PR DESCRIPTION
**Pull request by canova**
r=bholley in bugzilla

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix [Bug 1360488](https://bugzilla.mozilla.org/show_bug.cgi?id=1360488)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16819)

### CODE REVIEW

**Comment by canaltinova** [comment]
@bors-servo r-
(Needs coordinated landing. r-'ing because couldn't find anyone to push the m-c part. Feel free to land :) )

**Comment by bholley** [comment]
File a bug to get L3 access, and I'll vouch for you? https://www.mozilla.org/en-US/about/governance/policies/commit/

**Comment by canaltinova** [comment]
@bholley filed a bug, thanks!

**Comment by canaltinova** [comment]
Oops, forgot bindings.
@bors-servo r=bholley

**Comment by canaltinova** [comment]
@bors-servo r-
ugh, there are lots of PRs in the homu queue and I won't be here to watch this.

**Comment by emilio** [comment]
@bors-servo r-

Sorry @canaltinova, but I won't have time to land it right now :(.

File: components/style/stylist.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ pub struct Stylist {
     viewport_constraints: Option<ViewportConstraints>,
 
     /// If true, the quirks-mode stylesheet is applied.
-    quirks_mode: QuirksMode,
+    pub quirks_mode: QuirksMode,
```
Comment (emilio) [inline components/style/stylist.rs]: If you have the time, while you're here, could you make an accessor instead of just making the field public?

**Pull request merged**